rfcs/20211109-coral-support-for-tfjs-node.md

```diff
@@ -0,0 +1,173 @@
+headline in the release notes or blog post?
+
+## Demo Repository
+[This demo](https://github.com/mattsoulanille/node-tflite) is forked from [an external project that adds tflite support to node](https://github.com/seanchas116/node-tflite). The demo adds Coral support thorugh an [argument to the existing Napi bindings](https://github.com/mattsoulanille/node-tflite/blob/master/index.cc#L126-L138) and [links](https://github.com/mattsoulanille/node-tflite/blob/master/binding.gyp#L17) the [libedgetpu library](https://github.com/google-coral/libedgetpu) that is required for interacting with Coral devices.
+
+The demo has been tested on Linux X86 devices but has not been fully configured for Windows or Mac. It has also only been tested with a USB Coral device, although it should work with a PCIe device as well since it relies on libedgetpu for Coral support. To run the demo yourself, follow these steps:
+1. [Install the Edge TPU runtime](https://coral.ai/docs/accelerator/get-started#1-install-the-edge-tpu-runtime).
+2. In the root of the repository, install dependencies and compile the Napi bindings with `npm install`.
+3. In the `examples/electron-mediapipe-face` repository, run `npm install` to install dependencies.
+4. Run `npm start` to start the demo. You should see a screen like this:
+
+
+
+## Design Proposal
+
+### Overview
+We propose:
+1. To extend `tfjs-tflite` or `tfjs-node` to support running TFLite models in Node, or to write a new package that supports running tflite models in node.
+2. To provide a plugin system for TFLite Delegates and a Coral plugin that provides support for Coral accelerators.
+
+We exclude from the scope of this proposal:
+1. Coral support for web - This is of limited use for Coral devices, [which must be plugged in, unplugged, and plugged in again to work on the web](https://github.com/google-coral/webcoral#device-setup).
+2. Plugin support for web - This may be a future proposal.
+
+### High Level Constraints and Tradeoffs
+1. Package size - Constrained more than usual for an npm package since a target platform is IoT devices.
+2. Supported platforms vs Maintainability - TFLite does not ship precompiled binaries, and [Coral ships a limited set of precompiled binaries](https://github.com/google-coral/libedgetpu/releases/tag/release-grouper). We need to balance supported platforms with maintenance cost to the team. Ideally, we would let users compile TFLite and libedgetpu themselves for unsupported platforms.
+3. Where to implement the proposal - In tfjs-node? In tfjs-tflite? In a new package?
+4. Whether to use the [TFLite C api](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/c/c_api.h) or the [C++ api](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/BUILD#L343-L370). The demo uses the C api, but tfjs-tflite uses the C++ api. The APIs are nearly functionally equivalent. The C library is 2.9MB while the C++ library is 3.4MB.
+
+### Where to Implement
+
+#### In tfjs-node
```

Comment:
It will be good to explain what binaries would be shipped with the package, installation process and the similarity with tfjs-node.